### ISSUE TITLE
Intermitent failure on ConfigurationSetup#initializeConfiguration during livereload

### ISSUE THREAD

**Issue by loicmathieu** [quarkusio/quarkus#3868]
**Describe the bug**
Sometimes, the livereload crash after a change on one file with an error on the configuration build step: `Build step io.quarkus.deployment.steps.ConfigurationSetup#initializeConfiguration threw an exception: java.util.NoSuchElementException: Property quarkus.log.category."org.hibernate.SQL".level not found`

**Expected behavior**
Livereload should always works.

**Actual behavior**
After starting Quarkus in dev mode with `mvn clean compile quarkus:dev` and changing one file, livereload happens but crash. Restarting the apps via the same maven command solves the issue.

Here is the log of the issue:

```
2019-09-05 11:19:50,670 INFO  [io.qua.dev] (executor-thread-1) Changed source files detected, recompiling [...]
2019-09-05 11:19:52,513 INFO  [io.quarkus] (executor-thread-1) Quarkus stopped in 0.067s
2019-09-05 11:19:52,551 INFO  [io.qua.dep.QuarkusAugmentor] (executor-thread-1) Beginning quarkus augmentation
2019-09-05 11:19:52,703 INFO  [io.qua.ely.sec.dep.SecurityDeploymentProcessor] (build-20) Configuring from MPRealmConfig
2019-09-05 11:19:52,709 INFO  [io.qua.fly.FlywayProcessor] (build-5) Adding application migrations in path: [...]
2019-09-05 11:19:52,712 INFO  [io.qua.fly.FlywayProcessor] (build-5) Adding application migrations in path: [...]
2019-09-05 11:19:52,732 ERROR [io.qua.dev.DevModeMain] (executor-thread-1) Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.deployment.steps.ConfigurationSetup#initializeConfiguration threw an exception: java.util.NoSuchElementException: Property quarkus.log.category."org.hibernate.SQL".level not found
	at io.quarkus.runner.RuntimeRunner.run(RuntimeRunner.java:141)
	at io.quarkus.dev.DevModeMain.doStart(DevModeMain.java:180)
	at io.quarkus.dev.DevModeMain.restartApp(DevModeMain.java:195)
	at io.quarkus.dev.RuntimeUpdatesProcessor.doScan(RuntimeUpdatesProcessor.java:116)
	at io.quarkus.undertow.deployment.devmode.UndertowHotReplacementSetup.handleHotDeploymentRequest(UndertowHotReplacementSetup.java:72)
	at io.quarkus.undertow.deployment.devmode.UndertowHotReplacementSetup$1$1.handleRequest(UndertowHotReplacementSetup.java:61)
	at io.undertow.server.Connectors.executeRootHandler(Connectors.java:376)
	at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:830)
	at io.quarkus.runtime.CleanableExecutor$CleaningRunnable.run(CleanableExecutor.java:224)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2011)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1535)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1426)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:479)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.deployment.steps.ConfigurationSetup#initializeConfiguration threw an exception: java.util.NoSuchElementException: Property quarkus.log.category."org.hibernate.SQL".level not found
	at io.quarkus.builder.Execution.run(Execution.java:108)
	at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:121)
	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:115)
	at io.quarkus.runner.RuntimeRunner.run(RuntimeRunner.java:107)
	... 18 more
Caused by: java.util.NoSuchElementException: Property quarkus.log.category."org.hibernate.SQL".level not found
	at io.smallrye.config.SmallRyeConfig.getValue(SmallRyeConfig.java:89)
	at io.quarkus.deployment.steps.ConfigurationSetup.initializeConfiguration(ConfigurationSetup.java:225)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$1.execute(ExtensionLoader.java:768)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:415)
	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2011)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1535)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1426)
	... 2 more

2019-09-05 11:19:52,733 INFO  [io.qua.dev] (executor-thread-1) Hot replace total time: 2.080s 
```

**To Reproduce**
I cannot reliabily reproduce the error. It's an intermitent failure that happens on livereload.

**Configuration**
```properties
# Configure loggers
quarkus.log.category."com.mycompany.mydomain".level=DEBUG
%dev.quarkus.log.category."org.hibernate.SQL".level=DEBUG
```

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: Linux 5.0.0-25-generic #26-Ubuntu SMP Thu Aug 1 12:04:58 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
 - Output of `java -version`: openjdk version "11.0.4" 2019-07-16
 - Quarkus version or git rev: 0.21.1

**Comment by gsmet** [comment]
Can you try with the just released 0.21.2? We fixed a very similar issue.

**Comment by loicmathieu** [comment]
@gsmet I triggered more than 10 livereload and no issue with 0.21.2.
Sorry for the duplicate this is a bug I wanted to open since a few weeks and I didn't saw anything about it when I quickly looks at the list of bugfixes for 0.21.2.
